ext/ffi/lib.rs

```diff
@@ -351,10 +365,25 @@ struct ForeignFunction {
   result: NativeType,
 }
 
+#[derive(Deserialize, Debug)]
+#[allow(dead_code)]
```

Comment:
Why is this dead code?

Reply:
The type and name field for the constants are never actually read. I'll check if I can just mark the `ForeignStatic` as a zero-size variant.

Reply:
Oh ok, can you add a comment similar to the one at line 493 here?

Reply:
Added comment.

Reply:
Can't you just name the thing `_type` rather than `type_`? 
I think that would avoid the warning as well.